includes/enrolment/class-sensei-course-enrolment.php

```diff
@@ -358,6 +349,21 @@ private function store_enrolment_results( $user_id, Sensei_Course_Enrolment_Prov
 		}
 	}
 
+	/**
+	 * Helper to disable storing enrolment results when it isn't already stored and enrolment isn't provided.
+	 *
+	 * @param bool                                     $store_results      Whether to store the results.
+	 * @param int                                      $user_id            User ID.
+	 * @param int                                      $course_id          Course post ID.
+	 * @param bool                                     $had_existing_value True if a stale enrolment result is already stored.
+	 * @param Sensei_Course_Enrolment_Provider_Results $enrolment_results  Enrolment results object.
+	 *
+	 * @return bool
+	 */
+	public static function do_not_store_negative_enrolment_results( $store_results, $user_id, $course_id, $had_existing_value, $enrolment_results ) {
```

Comment:
Maybe change this to `should_store_negative_enrolment_results`?

Reply:
Aren't we filtering to not store negative enrolment results?

Reply:
Yes but this method returns true when we should store the results, right?

Reply:
For hooked methods it seems better to name the behavior it changes. Otherwise,
`add_filter( 'sensei_course_enrolment_store_results', [ Sensei_Course_Enrolment::class, 'should_store_negative_enrolment_results' ], 10, 5 );` doesn't really say what we're doing with the filter.